spec/domain/command_handlers/game_command_handler_spec.rb

```diff
@@ -116,231 +137,335 @@
               InstructionCard.u_turn(40),
               InstructionCard.u_turn(50)
             ]),
-            AllRobotsProgrammedEvent.new(id)
-          ]
-        end
-      end
-
-      context "when trying to program an instruction that is not dealt to player" do
-        before do
-          dispatch(create_game_command)
-          dispatch(start_game_command)
-
-          command.instruction_cards[0] = InstructionCard.move_3(790)
+          )
         end
 
-        specify do
-          expect { dispatch(command) }.to raise_error(IllegalInstructionCardError)
+        it 'denies re-programming the robot' do
+          expect { dispatch(command) }.to raise_error(RobotAlreadyProgrammedError)
         end
       end
 
-      context "when robot is already programmed" do
-        before do
-          dispatch(create_game_command)
-          dispatch(start_game_command)
-          dispatch(program_robot_command)
-        end
+      context 'given the player has not joined the game' do
+        before { command.player_id = steven }
 
-        specify do
-          expect { dispatch(command.dup) }.to raise_error(RobotAlreadyProgrammedError)
+        it 'denies programming the robot' do
+          expect { dispatch(command) }.to raise_error(PlayerNotInGameError)
         end
       end
+    end
 
-      context "when game not yet started" do
-        before { dispatch(create_game_command) }
-
-        specify do
-          expect { dispatch(command) }.to raise_error(GameNotRunningError)
-        end
+    context 'given an unstarted game' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.